### PR TITLE
ConsoleApp.DrinksInfo finally done! 

### PR DESCRIPTION
**Pull request by AhmadJer99**

### CODE REVIEW

File: DrinksInfo.AhmadJer99/DrinksInfo/Controllers/ProcessApiRequest.cs
Comment (chrisjamiecarter) [inline DrinksInfo.AhmadJer99/DrinksInfo/Controllers/ProcessApiRequest.cs:1]: 🟢 **Generic Method**

⭐

File: DrinksInfo.AhmadJer99/DrinksInfo/UserInterface/DrinkDetailsView.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+public class DrinkDetailsView
+{
+    private HttpClient? _httpClient;
+    private DrinkDetails? _drinkDetails;
+    public string DrinkId { get; set; }
+    public DrinkDetailsView(string drinkId)
+    {
+        DrinkId = drinkId;
+        var task = LoadDrinkDetailsById();
+        task.Wait();
```
Comment (chrisjamiecarter) [inline DrinksInfo.AhmadJer99/DrinksInfo/UserInterface/DrinkDetailsView.cs:16]: 🟠 **Avoid Synchronous Waiting on Async Methods**

💡 `task.Wait()` introduces a blocking call, which is not ideal. Instead, consider making the constructor `async` and calling `await LoadDrinkDetailsById()`.

File: DrinksInfo.AhmadJer99/DrinksInfo/UserInterface/DrinkDetailsView.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    private async Task LoadDrinkDetailsById()
+    {
+        var filter = "i=" + DrinkId;
+        var uriRequest = "https://www.thecocktaildb.com/api/json/v1/1/lookup.php?" + filter;
+
+        _httpClient = new HttpClient();
```
Comment (chrisjamiecarter) [inline DrinksInfo.AhmadJer99/DrinksInfo/UserInterface/DrinkDetailsView.cs:24]: 🟠 **HttpClient Instantiation**

💡 Creating a new `HttpClient` instance for every request is inefficient and can lead to socket exhaustion. Use a singleton or reuse a shared `HttpClient` instance.

File: DrinksInfo.AhmadJer99/README.md
Comment (chrisjamiecarter) [inline DrinksInfo.AhmadJer99/README.md:1]: 🟢 **Readme File**

⭐

**Review by chrisjamiecarter — APPROVED** [review]
Hey @AhmadJer99 👋,

Excellent work on your Drinks project submission 🎉!
I have performed a peer review and have added some praise and suggestions as bespoke comments.
Also, see below for some general feedback.

***

🟢  **Solution Structure**
Perfect submission and great solution structure.

🟢 **Style Guide Compliance**
Lovely code style.

🟢  **Console UI**
Lovely looking UI with great colour choices.

🟠 **User Experience**
I would suggest using `Spectre Console` for you selection menus to select from the list rather than manual user entry.

image

***

Feel free to reach out if you have any questions or if you'd like to collaborate further on any of these points 🆘 .

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter 👍

File: DrinksInfo.AhmadJer99/DrinksInfo/UserInterface/DrinkDetailsView.cs (hunk 1 above)
Comment (AhmadJer99) [reply DrinksInfo.AhmadJer99/DrinksInfo/UserInterface/DrinkDetailsView.cs:16]: I think I have tried doing that but I ran into other issues while trying to make the Ctor async , I will try again to do it by making the ctor or whatever function is calling async and await.
Thanks for the feedback !

File: DrinksInfo.AhmadJer99/DrinksInfo/UserInterface/DrinkDetailsView.cs (hunk 2 above)
Comment (AhmadJer99) [reply DrinksInfo.AhmadJer99/DrinksInfo/UserInterface/DrinkDetailsView.cs:24]: I still dont quite know what a singleton is , i will look into it tho and i will  try to fix that.

**Pull request closed without merging**